File: shared/utils/src/commonMain/kotlin/com/flowspeed/lib/util/PathValidator.kt
package com.flowspeed.lib.util

import com.flowspeed.lib.util.osfileutil.FileUtils
import java.io.File

object PathValidator {
    fun canWriteToThisPath(path: String): Boolean {
        return FileUtils.canWriteInThisFolder(path)
    }

    fun isValidPath(path: String): Boolean {
        if (path.isEmpty()) return false
        return runCatching {
            File(path).canonicalFile
            true
        }.getOrElse { false }
    }
}

File: FlowDownloadManager/src/Flow.Shared/Utils/PathValidator.cs
namespace Flow.Shared.Utils;

public static class PathValidator
{
    public static bool CanWriteToThisPath(string path)
    {
        return FileUtils.CanWriteInThisFolder(path);
    }

    public static bool IsValidPath(string path)
    {
        if (string.IsNullOrWhiteSpace(path)) return false;
        try
        {
            // GetFullPath throws if the path contains invalid characters or is malformed
            var fullPath = Path.GetFullPath(path);
            return true;
        }
        catch
        {
            return false;
        }
    }
}
